apps/sms/js/thread_ui.js

```diff
@@ -1387,26 +1389,38 @@ var ThreadUI = {
    * markup to support bidirectional content, but other panels still use it with
    * mozL10n.setAttributes as it would contain only localizable text. We should
    * get rid of this method once bug 961572 and bug 1011085 are landed.
-   * @param {string|{ id: string, args: Object }} content Should be either safe
-   * HTML string or object with l10nId and l10nArgs.
+   * @param {string|{ raw: string }|{ html: string }|
+   * {id: string, args: Object }} contentL10n Should be either safe HTML string
+   * or l10n properties or raw string.
    * @public
    */
-  setHeaderContent: function thui_setHeaderContent(content) {
-    if (typeof content === 'string') {
-      this.headerText.removeAttribute('data-l10n-id');
-      this.headerText.removeAttribute('data-l10n-args');
-
-      this.headerText.innerHTML = content;
-    } else {
+  setHeaderContent: function thui_setHeaderContent(contentL10n) {
+    if (typeof contentL10n === 'string') {
```

Comment:
Previously l10n guys advised to use approach [1] for ```setHeaderContent```, but patch was landed earlier, so I use the chance to unify this now. Also we already use the same approach in ```dialog.js```

[1] https://developer.mozilla.org/en-US/Firefox_OS/Developing_Gaia/localization_code_best_practices#Writing_APIs_that_operate_on_L10nIDs